core/block.ts

```diff
@@ -1104,16 +1104,24 @@ export class Block implements IASTNodeLocation, IDeletable {
   }
 
   /**
-   * Return all variables referenced by this block.
+   * Return id strings of all variables referenced by this block.
    *
    * @returns List of variable ids.
    */
   getVars(): string[] {
     const vars: string[] = [];
     for (let i = 0, input; (input = this.inputList[i]); i++) {
       for (let j = 0, field; (field = input.fieldRow[j]); j++) {
+        const variable = field.getVariable();
+        if (variable) {
+          vars.push(variable.getId());
+          continue;
+        }
+
+        // Backwards-compatible check used for fields that stored variable id
+        // as their value. Deprecated, implement Field.getVariable() instead.
+        // When this is removed, the continue keyword above can be removed too.
         if (field.referencesVariables()) {
```

Comment:
Can you add a call to `deprecation.warn` in places where `referencesVariables` is called as well?